File: admin/src/models/PersonalityModel.ts
export type Personality = {
  _id: string;
  name: string;
  description: string;
  images: Array<string>;
  primary_image: number;
  primary_image_blurhash: string;
  sight_link?: string;
  pdf: string;
  type: PersonalityType;
  index: number;
};

export enum PersonalityType {
  personalitate = "personalitate",
  legenda = "legenda",
}

File: lib/models/personality_model.dart
import 'package:visit_braila/utils/url_constants.dart';

class Personality {
  final String id;
  final String name;
  final String description;
  final List<String> images;
  final int primaryImage;
  final String primaryImageBlurhash;
  final String? sightLink;
  final String pdf;

  Personality({
    required this.id,
    required this.name,
    required this.description,
    required this.images,
    required this.primaryImage,
    required this.primaryImageBlurhash,
    required this.sightLink,
    required this.pdf,
  });

  factory Personality.fromJSON(Map<String, dynamic> json) {
    return Personality(
      id: json['_id'],
      name: json['name'],
      description: json['description'],
      images: List<String>.from(json['images'].map((image) => "$baseUrl$image")),
      primaryImage: json['primary_image'],
      primaryImageBlurhash: json['primary_image_blurhash'],
      sightLink: json['sight_link'],
      pdf: "$baseUrl${json['pdf']}",
    );
  }
}
